### ISSUE TITLE
Tensor.masked_scatter_ documentation does not have any information about `src` tensor broadcasting logic

### ISSUE THREAD

**Issue by wonjoolee95** [pytorch/pytorch#94353]
### 📚 The doc issue

The documentation for `Tensor.masked_scatter_` op at https://pytorch.org/docs/stable/generated/torch.Tensor.masked_scatter_.html#torch.Tensor.masked_scatter_ does not have information about the `src` tensor broadcasting logic. 

The example current PyTorch (nightly) behavior that shows the `src` tensor broadcasting:
```
>>> import torch
>>> self = torch.tensor([1, 1, 1, 1, 1])
>>> mask = torch.tensor([[0, 0, 0, 1, 1], [1, 1, 1, 1, 1], [1, 0, 1, 1, 0],[0, 1, 0, 0, 0],[0, 1, 0, 0, 0]])
>>> source = torch.tensor([[0, 0, 0, 0, 0], [0, 0, 0, 0, 0], [0, 0, 0, 0, 0],[0, 0, 0, 0, 0],[0, 0, 0, 0, 0]])
>>> self.mask_scatter(mask, source)
>>> self.masked_scatter(mask, source)
tensor([[1, 1, 1, 0, 0],
        [0, 0, 0, 0, 0],
        [0, 1, 0, 0, 1],
        [1, 0, 1, 1, 1],
        [1, 0, 1, 1, 1]])
```
As shown above, the `self` tensor broadcasts to the shape of `mask` tensor.  

### Suggest a potential alternative/fix

The documentation should have some information to note that the `self` tensor broadcasts to shape of the `mask` tensor. It should also describe additional broadcasting logic as well if possible. 

cc @svekars @carljparker @mikaylagawarecki
